yarn-project/simulator/src/public/executor.ts

```diff
@@ -71,6 +72,9 @@ async function executePublicFunctionAvm(
   );
   const worldStateJournal = new AvmPersistableStateManager(hostStorage);
   worldStateJournal.trace.accessCounter = executionContext.execution.callContext.sideEffectCounter;
+  for (const nullifier of executionContext.pendingNullifiers) {
+    worldStateJournal.nullifiers.cache.appendSiloed(nullifier.value);
+  }
```

Comment:
This is probably fine for now.... But if we plan to use this long-term (as opposed to having another layer of caching like the WorldStatePublicDB), maybe this should be a constructor argument to the state manager?